src/helpers/Constants.sol

```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+// Index marking the start of the data for the validation function.
```

Comment:
`RESERVED_VALIDATION_DATA_INDEX` is a hard coded cap fro preValidationHook limit, right? 
The doc above is confusing.

Reply:
Yes it is the size limit, but there is also another use for it. When providing data to each hook, the index is set to the index of the hook. But the data for the validation function (usually the signature) has this value `0xFF` instead of an index, regardless of the number of pre-validation hooks.

This value is used for both the "max size of pre validation hooks" and the "reserved (magic) index value for the validation function, not hooks". I'll update the comment to describe this better.

Reply:
Should we create two constants for that?

Reply:
We did it for `_1271_INVALID` and `_INTERFACE_ID_INVALID`.